stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1181,6 +1181,21 @@ public extension Tensor {
       let slice: Tensor = Raw.slice(self, begin: startIndices, size: boundSizes)
       return slice.squeezingShape(at: 0)
     }
+    @inline(__always)
+    set {
+      let orig = self[index]
+      // FIXME: Currently there's no existing non-in-place slice updating
+      // operator in TensorFlow. But since most scalar types are numeric, we do
+      // a hack by using arithmetic operations. We need to find a proper
+      // solution soon.
+      let diff = Tensor(handle: #tfop("Sub", newValue, orig))
+      let scatteredDiff = Raw.scatterNd(
+        indices: Tensor<Int32>(shape: [1, 1], scalars: [index]),
+        updates: Tensor([diff]),
```

Comment:
Maybe you could use `[diff]` directly without explicitly calling `Tensor.init`. Type inference for the `Scalar` type should work here.

Reply:
I tried that but I got this error:
```console
error: cannot convert value of type '[Tensor<Scalar>]' to expected argument type 'Tensor<_>'
```

Reply:
That's unfortunate. Thanks for trying!